File: Java/src/BAEKJOON/Num1018.java
/**
 * @author thelight0804<br>
 * Date: 2022.08.31<br>
 * Title: BAEKJOON 2784번 <br>
 * Project: 피보나치 수 2 <br>
 * Level: Bronze 5 <br>
 * 제출 시 class 이름을 Main으로 변경해야 한다
 */

package BAEKJOON;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Num1018{
  public static void main(String args[]) throws IOException {
    //num input
    BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
    int num = Integer.parseInt(bf.readLine());

    //memoization
    long memory[] = new long[91];
    memory[1] = 1;

    //Fibonacci
    for(int i=2;i<num+1;i++)
      memory[i] = memory[i - 1] + memory[i - 2];

    System.out.println(memory[num]);
  }
}

File: Python/BAEKJOON/Num1018.py
"""
Date:
  23.04.04
Problem:
  BAEKJOON 1018번
Project:
  체스판 다시 칠하기
Level:
  Silver 4
Author:
  thelight0804
"""

# standard_input = """8 8
# BWBWBWBW
# WBWBWBWB
# BWBWBWBW
# WBWBWBWB
# BWBWBWBW
# WBWBWBWB
# BWBWBWBW
# WBWBWBWB"""
# #12

#input n, m
n, m = map(int, input().split())
#input board
board = [input() for _ in range(n) ]
cnt = []

#cut 8*8 boards
for i in range(n-7):
  for j in range(m-7):
    white = 0
    black = 0
    #check W, B from row * col
    for k in range(i, i+8):
      for l in range(j, j+8):
        if (k + l) % 2 == 0: #Even
          if board[k][l] == 'W':
            black += 1
          if board[k][l] == 'B':
            white += 1
        else: #odd
          if board[k][l] == 'B':
            black += 1
          if board[k][l] == 'W':
            white += 1
    #add cnt
    cnt.append(white)
    cnt.append(black)

#output minimum
print(min(cnt))